File: src/player/property/avatar/avatars/Archer.java
package player.property.avatar.avatars;

import damage.Damage;
import graphics.ImageID;
import graphics.ImageFile;
import misc.math.game.GamePosition;
import player.LobbyPlayer;
import player.ServerGamePlayer;
import player.players.ArcherPlayer;
import player.property.avatar.Avatar;

public class Archer extends Avatar
{
	@Override public ImageID getIconImageID() // works
	{
		return ImageFile.ARCHER_ICON.getImageID();
	}

	@Override public String getDescription() { return "a funny fernkämpfer"; }

	@Override public ServerGamePlayer createServerGamePlayer(LobbyPlayer player, GamePosition position)
	{
		return new ArcherPlayer(player, position);
	}
}

File: cpp/src/player/property/avatar/avatars/Archer.cpp
#include "Archer.hpp"

#include <player/game/ArcherGamePlayer.hpp>
#include <player/property/avatar/AvatarID.hpp>

TextureID Archer::getIconTextureID() const
{
	return ARCHER_ICON_GID;
}

char Archer::getID() const
{
	return ARCHER_AID;
}

std::string Archer::getDescription() const
{
	return "legolas; enough said";
}
